### ISSUE TITLE
Some complex machine learning code. Task 2.3 gen_ai lab

### ISSUE THREAD

**Issue by J-Doza** [onasito/Group-5#98]
This code has some documentation but some of the math is hard to follow:

import pandas as pd
import numpy as np
from sklearn.model_selection import KFold

columns = ['cylinders', 'displacement', 'horsepower', 'weight', 'acceleration', 'model year', 'origin']
data = pd.read_csv("auto-mpg2.csv", header=None, sep=',')
y = data.iloc[:, 0]  # mpg data
X = data.drop(columns=[0, 8], axis=1)  # columns 0 and 8 are "mpg", and "car name" respectively
X.columns = columns

# Standardize the data
X_standardized = (X - X.mean()) / X.std()
X_standardized = pd.DataFrame(X_standardized, columns=X.columns)

def Linear_Regression(X, y):
    X_inter = np.c_[np.ones((X.shape[0], 1)), X]
    coefficients = np.dot(np.dot(np.linalg.inv(np.dot(X_inter.transpose(), X_inter)), X_inter.transpose()), y)
    return coefficients

def RMSE(y_pred, y_true, n):
    return np.sqrt(np.sum((y_pred - y_true) ** 2) / n)

RMSE_list = []
KF = KFold(n_splits=10, shuffle=True, random_state=13)
fold_counter = 1 # to keep track of which fold we're in
columns += ['RMSE'] # Add the column header for RMSE
for header in columns: print('\t\t', header, end=' ')
print()
for train_index, test_index in KF.split(X_standardized):
    X_train  = X_standardized.iloc[train_index]
    X_test   = X_standardized.iloc[test_index]
    y_train  = y.iloc[train_index]
    y_test   = y.iloc[test_index]

    # Use our Linear Regression equation
    b = Linear_Regression(X_train, y_train)
    y_pred = np.dot(np.c_[np.ones((X_test.shape[0], 1)), X_test], b)

    # Use our RMSE calc and add value to our list
    RMSE_vals = RMSE(y_pred, y_test, len(y_test))
    RMSE_list.append(RMSE_vals)

    print(f"Fold: {fold_counter}")
    print("Coefficients:", b[1:], RMSE_list[fold_counter - 1], "\n")
    fold_counter += 1

average_RMSE = np.mean(RMSE_list)
print(f"Average RMSE of all folds: {average_RMSE}")

I'm going to ask chatgpt to send with understanding the more complex functions.

**Comment by J-Doza** [comment]
pull request #99 solves this issue.
